src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -69,6 +71,7 @@ public abstract class BatchExportProcessor<T> : BaseExportProcessor<T>
             Guard.ThrowIfOutOfRange(exporterTimeoutMilliseconds, min: 0);
 
             this.circularBuffer = new CircularBuffer<T>(maxQueueSize);
+            this.circularBufferDroppedCount = Sdk.SdkInternalStateMeter.CreateObservableCounter($"{typeof(T)}.CircularBufferDroppedCount", () => this.DroppedCount);
```

Comment:
check if we need to read this via Interlock.Read

Reply:
that should be fixed with https://github.com/open-telemetry/opentelemetry-dotnet/issues/4226?